examples/cs/core/guid/program.cs

```diff
@@ -17,6 +19,24 @@ public static string Convert(Guid value, string unused)
         {
             return value.ToString();
         }
+
+        #endregion
+
+        #region Guid-blob conversion
+
+        public static Guid Convert(ArraySegment<byte> value, Guid unused)
+        {
+            var bits = new byte[16];
```

Comment:
We can do this without an allocation.

1. Check `value.Count == 16` otherwise throw something like `InvalidDataException`.
1. Extract the bytes themselves and use the `Guid(int, short, short, byte, ..., byte)` constructor:

```
    int a =
          ((int)value.Array[value.Offset + 3] << 24)
        | ((int)value.Array[value.Offset + 2] << 16)
        | ((int)value.Array[value.Offset + 1] <<  8)
        | value.Array[value.Offset + 0];
    short b = /* similar for the next two bytes */
    short c = /* similar for the next two bytes */
    return new Guid(a, b, c,
        value.Array[value.Offset + 8],
        ...
        value.Array[value.Offset + 15]);
```